frontend/src/components/c-ramp.vue

```diff
@@ -111,29 +121,31 @@ export default {
       if (this.isDeletesContribution(slice)) {
         return this.deletesContributionRampSize;
       }
-      const newSize = 100 * (slice.insertions / this.avgsize);
+      // '+' unary operator here attempts to convert this.avgsize to number, if it is not already one
+      const newSize = 100 * (slice.insertions / +this.avgsize);
       return Math.max(newSize * this.rampSize, 0.5);
     },
-    getContributionMessage(slice, commit) {
-      let title = '';
-      if (this.tframe === 'commit') {
+    // commit of type CommitResult must be provided if tframe === 'commit', only pass null if tframe !== 'commit
+    getContributionMessage(slice: Commit, commit: CommitResult | null) {
+      if (this.tframe === 'commit' && commit !== null) {
         return `[${slice.date}] ${commit.messageTitle}: +${commit.insertions} -${commit.deletions} lines `;
       }
 
-      title = this.tframe === 'day'
-            ? `[${slice.date}] Daily `
-            : `[${slice.date} till ${slice.endDate}] Weekly `;
+      let title = this.tframe === 'day'
+          ? `[${slice.date}] Daily `
+          : `[${slice.date} till ${slice.endDate}] Weekly `;
       title += `contribution: +${slice.insertions} -${slice.deletions} lines`;
       return title;
     },
-    openTabZoom(user, slice, evt) {
+    // Only to be used when this.tframe !== 'Commit'
+    openTabZoom(user: User, slice: Commit, evt: MouseEvent) {
       // prevent opening of zoom tab when cmd/ctrl click
       if (window.isMacintosh ? evt.metaKey : evt.ctrlKey) {
         return;
       }
 
       const zoomUser = { ...user };
-      zoomUser.commits = user.dailyCommits;
+      zoomUser.commits = user.dailyCommits as Commit[];
```

Comment:
Not sure if this is a safe cast since `DailyCommit` doesn't have the `insertions` and `deletions` fields. Will this break anything in the zoom view? (since if that file tries to access these properties, it will just be undefined).
Understand that this is existing behaviour and not introduced by this PR, but if there is an existing bug because of this, we should file an issue & fix that

Reply:
Similarly, due to the unknown nature of the `openTabZoom` method, this seemingly unsafe cast may not even matter. Seems like removing the statement entirely does not affect the behavior of the program, at least on my Windows OS.

Reply:
Refer to [this comment](https://github.com/reposense/RepoSense/pull/2037#discussion_r1341945935) for solution to the unsafe casting